imdclient/IMDServer.py

```diff
@@ -0,0 +1,357 @@
+
+            self._conn.settimeout(5)
+
+            self._expect_header(IMDHeaderType.IMD_GO)
+
+            self._conn.settimeout(read_timeout)
+        except Exception as e:
+            self.stop()
+            raise e
+
+    def write_frame(self, imdframe):
+
+        if self.sinfo.time:
+            if imdframe.time is None:
+                raise ValueError(
+                    f"IMDServer: Time enabled for session, but frame does not contain time information"
+                )
+
+            struct.pack_into(
+                self._time_fmt,
+                self._time_view,
+                0,
+                imdframe.dt,
+                imdframe.time,
+                imdframe.step,
+            )
+
+        if self.sinfo.box:
+            if imdframe.box is None:
+                raise ValueError(
+                    f"IMDServer: Box enabled for session, but frame does not contain box information"
+                )
+
+            src = np.ascontiguousarray(imdframe.box, dtype=self._array_dtype)
+            target = np.frombuffer(self._box_view, dtype=self._array_dtype)
+            target = target.reshape(src.shape)
+            np.copyto(target, src)
+
+        if self.sinfo.positions:
+            if imdframe.positions is None:
+                raise ValueError(
+                    f"IMDServer: Positions enabled for session, but frame does not contain positions information"
+                )
+
+            src = np.ascontiguousarray(
+                imdframe.positions, dtype=self._array_dtype
+            )
+            target = np.frombuffer(self._pos_view, dtype=self._array_dtype)
+            target = target.reshape(src.shape)
+            np.copyto(target, src)
+
+        if self.sinfo.velocities:
+            if imdframe.velocities is None:
+                raise ValueError(
+                    f"IMDServer: Velocities enabled for session, but frame does not contain velocities information"
+                )
+
+            src = np.ascontiguousarray(
+                imdframe.velocities, dtype=self._array_dtype
+            )
+            target = np.frombuffer(self._vel_view, dtype=self._array_dtype)
+            target = target.reshape(src.shape)
+            np.copyto(target, src)
+
+        if self.sinfo.forces:
+            if imdframe.forces is None:
+                raise ValueError(
+                    f"IMDServer: Forces enabled for session, but frame does not contain forces information"
+                )
+
+            src = np.ascontiguousarray(
+                imdframe.forces, dtype=self._array_dtype
+            )
+            target = np.frombuffer(self._force_view, dtype=self._array_dtype)
+            target = target.reshape(src.shape)
+            np.copyto(target, src)
+
+        self._conn.sendall(self._send_buf)
+
+    def _alloc_send_buf(self):
+
+        buf_len = 0
+        xvf_size = 12 * self.n_atoms
+        if self.sinfo.time:
+            buf_len += IMDHEADERSIZE + IMDTIMEPACKETLENGTH
```

Comment:
```suggestion
        buf_len += int(self.sinfo.time) * (IMDHEADERSIZE + IMDTIMEPACKETLENGTH)
```

Reply:
This maybe an issue since `sinfo.time >=1` as per the protocol. `int(bool(self.sinfo.time))` might be a better option
```suggestion
        buf_len += int(bool(self.sinfo.time)) * (IMDHEADERSIZE + IMDTIMEPACKETLENGTH)
```